src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -975,12 +1032,50 @@ namespace
             }
 
             // Insert EDF.
-            if (rend_params.envMap->IsSubClassOf(AppleseedEnvMap::get_class_id()))
+            if (has_appleseed_sky_environment(rend_params))
             {
                 auto appleseed_envmap = static_cast<AppleseedEnvMap*>(rend_params.envMap);
                 if (appleseed_envmap)
                 {
-                    scene.environment_edfs().insert(appleseed_envmap->create_envmap(env_edf_name.c_str()));
+                    auto env_map = appleseed_envmap->create_envmap(env_edf_name.c_str());
+                    
+                    INode* sun_node(nullptr);
+                    BOOL sun_node_on(false);
+                    appleseed_envmap->GetParamBlock(0)->GetValueByName(_T("sun_node"), time, sun_node, FOREVER);
+                    appleseed_envmap->GetParamBlock(0)->GetValueByName(_T("sun_node_on"), time, sun_node_on, FOREVER);
+
+                    float sun_theta, sun_phi;
+                    if (sun_node && sun_node_on)
+                    {
+                        Matrix3 sun_transform = sun_node->GetObjTMAfterWSM(time);
+                        sun_transform.NoTrans();
+                        Point3 sun_dir = (Point3::ZAxis * sun_transform).Normalize();
+
+                        sun_theta = std::acosf(sun_dir.z);
+
+                        float cos_phi = sun_dir.x / sqrtf(1.0f - std::powf(sun_dir.z, 2));
+                        
+                        if (cos_phi > 1.0f)
```

Comment:
Can't it also be smaller than -1.0f? If that's the case you could simply do

    cos_phi = asf::clamp(cos_phi, -1.0f, 1.0f);